spec/lib/booker/v4.1/customer_spec.rb

```diff
@@ -273,6 +274,40 @@
     end
   end
 
+  describe '#specials' do
+    let(:expected_params) {described_class::DEFAULT_PAGINATION_PARAMS.merge({
+                                                                              access_token: 'access_token',
+                                                                              LocationID: booker_location_id
+                                                                            })}
+
+    before do
+      expect(client).to receive(:access_token).and_return 'access_token'
+      expect(client).to receive(:paginated_request).with(
+        method: :post,
+        path: '/v4.1/customer/specials',
+        params: expected_params,
+        model: Booker::V4::Models::Special,
+        fetch_all: true
+      ).and_return([])
+    end
+
+    it 'delegates to get_booker_resources' do
+      expect(client.specials(location_id: booker_location_id)).to eq []
+    end
+
+    context 'other params' do
+      let(:expected_params) {described_class::DEFAULT_PAGINATION_PARAMS.merge({
+                                                                                access_token: 'access_token',
+                                                                                LocationID: booker_location_id,
+                                                                                another_option: 'foo'
+                                                                              })}
+
+      it 'merges passed in params into base params' do
+        expect(client.specials(location_id: booker_location_id, params: {another_option: 'foo'})).to eq []
+      end
```

Comment:
shouldn't this be `expected_params`?

Reply:
nvm I see what's happening. I find it a little difficult to read when the expect is in the before block. It's not very clear what the test is testing based on the actual lines in the test. I know this is a pattern we use though so it's fine, just confused me.